File: common/src/main/java/com/insurtech/kanguro/common/enums/ClaimStatus.kt
package com.insurtech.kanguro.common.enums

import androidx.annotation.StringRes
import com.insurtech.kanguro.common.R

enum class ClaimStatus(@StringRes val value: Int, val stringValue: String) {
    Submitted(R.string.claim_status_submitted, "Submitted"),
    Assigned(R.string.claim_status_assigned, "Assigned"), // Created based on UI, should match backend
    Approved(R.string.claim_status_approved, "Approved"),
    InReview(R.string.claim_status_in_review, "InReview"), // Created based on UI, should match backend
    Draft(R.string.claim_status_draft, "Draft"),
    Closed(R.string.claim_status_closed, "Closed"),
    Deleted(R.string.claim_status_deleted, "Deleted"),
    Denied(R.string.claim_status_denied, "Denied"),
    Paid(R.string.claim_status_approved_paid, "Paid"),
    Unknown(R.string.claim_status_unknown, "Unknown"),
    PendingMedicalHistory(R.string.claim_status_pending_medical_history, "PendingMedicalHistory"),
    MedicalHistoryInReview(R.string.claim_status_medical_history_in_review, "MedicalHistoryInReview");

    companion object {
        fun fromString(value: String): ClaimStatus {
            return values().firstOrNull { it.stringValue == value } ?: Unknown
        }
    }
}

File: iosApp/Modules/KanguroSharedDomain/Sources/KanguroSharedDomain/Claim/Models/ClaimStatus.swift
public enum ClaimStatus: String, Equatable {

    case Submitted
    case Assigned
    case InReview
    case Approved
    case Draft
    case Closed
    case Deleted
    case Denied
    case Paid
    case PendingMedicalHistory
    case MedicalHistoryInReview
    case Unknown

    public var index: Int {
        switch self {
        case .Submitted:
            return 1
        case .Assigned:
            return 2
        case .InReview:
            return 3
        case .Closed:
            return 4
        default:
            return 0
        }
    }
}

public extension ClaimStatus {
    init(value: String) {
        switch value.lowercased() {
        case "submitted": self = .Submitted
        case "assigned": self = .Assigned
        case "inreview" : self = .InReview
        case "approved" : self = .Approved
        case "draft" : self = .Draft
        case "closed" : self = .Closed
        case "deleted" : self = .Deleted
        case "denied" : self = .Denied
        case "paid" : self = .Paid
        case "pendingmedicalhistory" : self = .PendingMedicalHistory
        case "medicalhistoryinreview" : self = .MedicalHistoryInReview
        default: self = .Unknown
        }
    }
}
